### ISSUE TITLE
ResumeNER数据集上不收敛

### ISSUE THREAD

**Issue by luosmart** [jiesutd/NCRFpp#70]
我使用LatticeLSTM项目中您提供的ResumeNER数据集，训练了多次，结果都是不收敛，请问可能是什么问题？多谢您啊！以下是我的日志记录：
```
Seed num: 42
MODEL: train
Load pretrained char embedding, norm: False, dir: ResumeNER/gigaword_chn.all.a2b.uni.ite50.vec
Embedding:
     pretrain word:11327, prefect match:63, case_match:0, oov:76, oov%:0.542857142857
Training model...
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
DATA SUMMARY START:
 I/O:
     Tag          scheme: BMES
     MAX SENTENCE LENGTH: 250
     MAX   WORD   LENGTH: -1
     Number   normalized: True
     Word  alphabet size: 1895
     Char  alphabet size: 140
     Label alphabet size: 29
     Word embedding  dir: None
     Char embedding  dir: ResumeNER/gigaword_chn.all.a2b.uni.ite50.vec
     Word embedding size: 50
     Char embedding size: 50
     Norm   word     emb: False
     Norm   char     emb: False
     Train  file directory: ResumeNER/train.char.bmes
     Dev    file directory: ResumeNER/dev.char.bmes
     Test   file directory: ResumeNER/test.char.bmes
     Raw    file directory: None
     Dset   file directory: None
     Model  file directory: ResumeNER/lstmcrf
     Loadmodel   directory: None
     Decode file directory: None
     Train instance number: 3821
     Dev   instance number: 463
     Test  instance number: 477
     Raw   instance number: 0
     FEATURE num: 0
 ++++++++++++++++++++++++++++++++++++++++
 Model Network:
     Model        use_crf: True
     Model word extractor: LSTM
     Model       use_char: True
     Model char extractor: CNN
     Model char_hidden_dim: 50
 ++++++++++++++++++++++++++++++++++++++++
 Training:
     Optimizer: SGD
     Iteration: 1000
     BatchSize: 64
     Average  batch   loss: True
 ++++++++++++++++++++++++++++++++++++++++
 Hyperparameters:
     Hyper              lr: 0.1
     Hyper        lr_decay: 0.05
     Hyper         HP_clip: None
     Hyper        momentum: 0.0
     Hyper              l2: 1e-08
     Hyper      hidden_dim: 200
     Hyper         dropout: 0.5
     Hyper      lstm_layer: 1
     Hyper          bilstm: True
     Hyper             GPU: True
DATA SUMMARY END.
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
build network...
use_char:  True
char feature extractor:  CNN
word feature extractor:  LSTM
use crf:  True
build word sequence feature extractor: LSTM...
build word representation...
build char sequence feature extractor: CNN ...
build CRF...
Epoch: 0/1000
 Learning rate is set as: 0.1
     Instance: 3821; Time: 37.96s; loss: 4187.0237; acc: 44252.0/124099.0=0.3566
Epoch: 0 training finished. Time: 37.96s, speed: 100.66st/s,  total loss: 4187.02367401
totalloss: 4187.02367401
gold_num =  1497  pred_num =  487  right_num =  0
Dev: time: 1.41s speed: 333.80st/s; acc: 0.3261
Exceed previous best acc score: -10
Save current best model in file: ResumeNER/lstmcrf.0.model
gold_num =  1630  pred_num =  496  right_num =  5
Test: time: 1.60s, speed: 302.52st/s; acc: 0.3086
Epoch: 1/1000
 Learning rate is set as: 0.0952380952381
     Instance: 3821; Time: 36.68s; loss: 1955.0515; acc: 49655.0/124099.0=0.4001
Epoch: 1 training finished. Time: 36.68s, speed: 104.18st/s,  total loss: 1955.05152512
totalloss: 1955.05152512
gold_num =  1497  pred_num =  806  right_num =  70
Dev: time: 1.40s speed: 334.84st/s; acc: 0.4839
Exceed previous best acc score: 0.326133909287
Save current best model in file: ResumeNER/lstmcrf.1.model
gold_num =  1630  pred_num =  827  right_num =  65
Test: time: 1.60s, speed: 302.06st/s; acc: 0.4795
Epoch: 2/1000
 Learning rate is set as: 0.0909090909091
     Instance: 3821; Time: 36.42s; loss: 1717.9032; acc: 49253.0/124099.0=0.3969
Epoch: 2 training finished. Time: 36.42s, speed: 104.93st/s,  total loss: 1717.90322876
totalloss: 1717.90322876
gold_num =  1497  pred_num =  443  right_num =  0
Dev: time: 1.41s speed: 332.91st/s; acc: 0.4477
gold_num =  1630  pred_num =  449  right_num =  0
Test: time: 1.60s, speed: 302.09st/s; acc: 0.4432
Epoch: 3/1000
 Learning rate is set as: 0.0869565217391
     Instance: 3821; Time: 34.46s; loss: 1591.0625; acc: 49245.0/124099.0=0.3968
Epoch: 3 training finished. Time: 34.46s, speed: 110.87st/s,  total loss: 1591.06254387
totalloss: 1591.06254387
```
谢谢！

**Comment by luosmart** [comment]
#71 fix this problem.

**Comment by jiesutd** [comment]
Hi @luosmart , in addition to the problem you raised, there are several issues to be noticed:
1. As the ResumeNER data is based on Chinese character sequence. In this situation, the `word` in NCRF++ means the Chinese character, and the `character` in NCRF++ is useless (as the basic input unit is already the character rather than the word). So you can set `use_char=False`

2.  The learning rate lr seems too large, with the large batch size (64 in your experiment), the training will not be stable in some datasets. If you find the unstable training process, you may consider using a small lr or batch size. Of course, you can tune your model based on the model performance.

3. Based on my experience, when training the Chinese NER model, the `norm_word_emb` has a big impact. You can try both `norm_word_emb=False` or `norm_word_emb=True`.

**Comment by luosmart** [comment]
@jiesutd Thank you for the detail.
